client/my-sites/themes/themes-selection.jsx

```diff
@@ -216,7 +219,7 @@ export const ConnectedThemesSelection = connect(
 		// results and sends all of the themes at once. QueryManager is not expecting such behaviour
 		// and we ended up loosing all of the themes above number 20. Real solution will be pagination on
 		// Jetpack themes endpoint.
-		const number = ! [ 'wpcom', 'wporg' ].includes( sourceSiteId ) ? 2000 : 30;
+		const number = ! [ 'wpcom', 'wporg' ].includes( sourceSiteId ) ? 2000 : 100;
```

Comment:
> Because our non-paginated queries only load the first pages' worth of themes into state (even if the query itself returns several more)

Am I reading this right, that state might only be populated with a percentage of themes returned by the API, not always _all_ themes returned from the API?..

Reply:
Correct! Until you click on All Themes to trigger a paginated query, the state will only be populated with a fixed number of themes, controlled by that number in the code above.

Reply:
Hm, so it sounds like there's still a "real" solution to come; as long as that's tracked somewhere, let's go with this to not block progress.